File: src/test/java/store/wckd/server/factory/UserFactory.java
package store.wckd.server.factory;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;
import store.wckd.server.entity.User;
import store.wckd.server.repository.UserRepository;

public class UserFactory implements Factory<User> {
    private final UserRepository userRepository;

    public UserFactory(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public Flux<User> createMany(int amount) {
        //noinspection ReactiveStreamsNullableInLambdaInTransform
        return Flux
                .range(1, amount)
                .map(id -> createOne().block());
    }

    @Override
    public Mono<User> createOne() {
        return Mono.just(userRepository.save(new User(
                0L,
                "fake username",
                "fake email",
                "fake password"
        )));
    }
}

File: src/test/java/store/wckd/server/factory/UserFactory.kt
package store.wckd.server.factory

import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import store.wckd.server.entity.User
import store.wckd.server.repository.UserRepository

class UserFactory(private val userRepository: UserRepository) : Factory<User> {
    override fun createMany(amount: Int): Flux<User> {
        return Flux
                .range(1, amount)
                .map { createOne().block()!! }
    }

    override fun createOne(): Mono<User> =
            Mono.just(userRepository.save(User(
                    username = "fake username",
                    email = "fake email",
                    password = "fake password"
            )))

}